packages/pigeon/lib/swift_generator.dart

```diff
@@ -1531,8 +1536,26 @@ private func nilOrValue<T>(_ value: Any?) -> T? {
 
     final String varChannelName = '${name}Channel';
     addDocumentationComments(indent, documentationComments, _docCommentSpec);
-    indent.writeln(
-        'let $varChannelName = FlutterBasicMessageChannel(name: "$channelName", binaryMessenger: binaryMessenger, codec: codec)');
+    String? taskQueue;
+    if (taskQueueType != TaskQueueType.serial) {
```

Comment:
im confused here - does `TaskQueueType.serial` mean it's not background queue? If it means it's a main queue, maybe we should rename to `serialMainQueue` instead.

Reply:
Doc states:

```dart
  /// Handlers are invoked serially on the default thread. This is the value if
  /// unspecified.
```

Reply:
> im confused here - does `TaskQueueType.serial` mean it's not background queue?

No, it means that all calls on that API object using the serial task queue will be executed in order, each one completing before the next starts.

As compared with, e.g., a default `NSOperationQueue`, which will dispatch jobs into a thread pool, and might run everything in parallel on N different threads.

Reply:
> Handlers are invoked serially on the default thread.

Is the "default thread" referring to main thread?

Reply:
No, it's an arbitrary background thread. That's why the API called by the generated code is `makeBackgroundTaskQueue`.

Calling things serially on the main thread is what platform channels do by default, and what they always did before the creation of the task queue API. The whole point of adding task queues to the channel APIs in the engine was to move them off of the main thread.

Reply:
Please forgive my numerous questions, I think I may be missing something. 

What you just described sounds very like the `.serialBackgroundThread` case, rather than the `.serial` case. Also, the code here is: 

```
if NOT serial { // the only other case is serialBackgroundThread
  use makeBackgroundTaskQueue API
}
```

Reply:
> What you just described sounds very like the `.serialBackgroundThread` case, rather than the `.serial` case.

Sorry, I forgot that Pigeon for some reason has a task queue enum value that actually means "don't use task queues". I have no idea why it was written that way; it's incredibly confusing.

So yes, it's the main thread and everything I said was wrong.

> maybe we should rename to `serialMainQueue` instead

We should probably just remove it, or if we keep it call it the even simpler `none`, but either way that's out of scope here.

Reply:
Gotcha. Thanks! 

@feduke-nukem while you are at it, maybe we can add an example usage in README? This should help clear up this kind of confusion for other people. What do you think? https://github.com/flutter/flutter/issues/162034

Reply:
Sure @hellohuanlin